src/status_im/hardwallet/core.cljs

```diff
@@ -49,17 +110,156 @@
+        pairing (get-pairing db)]
+    {:hardwallet/unpair-and-delete {:pin     pin
+                                    :pairing pairing}}))
+
+(fx/defn on-delete-success
+  [{:keys [db] :as cofx}]
+  (fx/merge cofx
+            {:db                              (-> db
+                                                  (assoc-in [:hardwallet :secrets] nil)
+                                                  (assoc-in [:hardwallet :application-info] nil)
+                                                  (assoc-in [:hardwallet :on-card-connected] nil)
+                                                  (assoc-in [:hardwallet :pin] {:status      nil
+                                                                                :error-label nil
+                                                                                :on-verified nil}))
+             :hardwallet/get-application-info nil
+             :utils/show-popup                {:title   ""
+                                               :content (i18n/label :t/card-reseted)}}
+            (remove-pairing-from-account)
+            (navigation/navigate-to-cofx :keycard-settings nil)))
+
+(fx/defn on-delete-error
+  [{:keys [db] :as cofx} error]
+  (log/debug "[hardwallet] delete error" error)
+  (fx/merge cofx
+            {:db                              (-> db
+                                                  (assoc-in [:hardwallet :on-card-connected] nil)
+                                                  (assoc-in [:hardwallet :pin] {:status      nil
+                                                                                :error-label nil
+                                                                                :on-verified nil}))
+             :hardwallet/get-application-info nil
+             :utils/show-popup                {:title   ""
+                                               :content (i18n/label :t/something-went-wrong)}}
+            (navigation/navigate-to-cofx :keycard-settings nil)))
+
+(fx/defn reset-card-pressed [cofx]
+  (navigation/navigate-to-cofx cofx :reset-card nil))
+
+(fx/defn delete-card
+  [{:keys [db] :as cofx}]
+  (let [keycard-instance-uid (get-in db [:hardwallet :application-info :instance-uid])
+        account-instance-uid (get-in db [:account/account :keycard-instance-uid])]
+    (if (or (nil? account-instance-uid)
+            (and keycard-instance-uid
+                 (= keycard-instance-uid account-instance-uid)))
+      {:hardwallet/delete nil}
+      (unauthorized-operation cofx))))
+
+(fx/defn reset-card-next-button-pressed
+  [{:keys [db] :as cofx}]
+  (let [card-connected? (get-in db [:hardwallet :card-connected?])
+        puk-retry-counter (get-in db [:hardwallet :application-info :puk-retry-counter])
+        pin-retry-counter (get-in db [:hardwallet :application-info :pin-retry-counter])
+        enter-step (if (zero? pin-retry-counter) :puk :current)]
```

Comment:
correct me if i'm wrong, but as i understand it will be always `:current` as there is another `if (zero? puk-retry-counter)` form later

Reply:
`enter-step` depends on whether `pin-retry-counter` is 0 or not, but later if checks `puk-retry-counter` which is another variable (if i understand your question properly)